lib/SILOptimizer/Utils/CanonicalizeOSSALifetime.cpp

```diff
@@ -287,16 +287,13 @@ void CanonicalizeOSSALifetime::extendLivenessToDeinitBarriers() {
     }
   }
   for (auto *edge : barriers.edges) {
-    if (edge == getCurrentDef()->getParentBlock()) {
-      // A destroy should be inserted at the begin of this block, but that does
-      // not require adding any instructions to liveness.
-      continue;
-    }
     auto *predecessor = edge->getSinglePredecessorBlock();
     assert(predecessor);
     liveness->updateForUse(&predecessor->back(),
                            /*lifetimeEnding*/ false);
   }
+  // Ignore barriers.initialBlocks.  Reachability reached the beginning of these
+  // blocks, but no instructions need to be added to liveness.
```

Comment:
>but no instructions need to be added to liveness.

This sounds rather mysterious to me. The simple fact is the no barriers were found on this path so there's obviously no need to extend liveness.

Reply:
Attempted to dispel the mystery by rephrasing as
```
  // Ignore barriers.initialBlocks.  If the collection is non-empty, it
  // contains the def-block.  Its presence means that no barriers were found
  // between lifetime ends and def.  In that case, no new instructions need to
  // be added to liveness.
```